### ISSUE TITLE
Clicking on OptionButton incorrectly triggers NOTIFICATION_WM_WINDOW_FOCUS_OUT / NOTIFICATION_WM_WINDOW_FOCUS_IN

### ISSUE THREAD

**Issue by mouring** [godotengine/godot#60050]
### Godot version

4.0 Alpha 6

### System information

MacOS 12.3 on Intel MacBook Pro

### Issue description

When using the Window Focus detection via _notification(..) to pause and restart the music when leaving/returning to the window it also pauses the music when you interact with an OptionButton.

This was not the case with 3.4.x.

### Steps to reproduce

Create a simple project with a simple scene with an OptionButton, AudioStreamPlayer2D, and set the audioplayer to a WAV or MP3 and autoplay.

Attach a script and include:
```
var pausePositon
func _notification(what) -> void:
	match what:
		NOTIFICATION_WM_WINDOW_FOCUS_IN:
			$AudioStreamPlayer2D.play(pausePositon)
		NOTIFICATION_WM_WINDOW_FOCUS_OUT:
			pausePositon = $AudioStreamPlayer2D.get_playback_position()
			$AudioStreamPlayer2D.stop()
```

Save and run the scene.

**Comment by mouring** [comment]
It will not let me attach the example zip file.  It can be found at: https://www.dropbox.com/s/lqzlku0u4wbqfck/example.zip?dl=0

**Comment by sriramun** [comment]
`NOTIFICATION_APPLICATION_FOCUS_IN` and `NOTIFICATION_APPLICATION_FOCUS_OUT` are more appropriate notifications to use here, I think.

Relevant docs: https://docs.godotengine.org/en/latest/classes/class_node.html#constants

> NOTIFICATION_WM_WINDOW_FOCUS_IN = 1004 --- Notification received from the OS when the node's parent [Window](https://docs.godotengine.org/en/latest/classes/class_window.html#class-window) is focused. This may be a change of focus between two windows of the same engine instance, or from the OS desktop or a third-party application to a window of the game (in which case [NOTIFICATION_APPLICATION_FOCUS_IN](https://docs.godotengine.org/en/latest/classes/class_node.html?highlight=NOTIFICATION_WM_WINDOW_FOCUS_IN#class-node-constant-notification-application-focus-in) is also emitted).

> NOTIFICATION_APPLICATION_FOCUS_IN = 2016 --- Notification received from the OS when the application is focused, i.e. when changing the focus from the OS desktop or a thirdparty application to any open window of the Godot instance.

**Comment by mouring** [comment]
Thank you for your suggestion.   3.x didn't have NOTIFICATION_APPLICATION_FOCUS_*, and it is nice to have this separation.

However, this is still a bug for NOTIFICATION_WM_WINDOW_FOCUS_* as per the documentation.  As it states this should be triggered between window changes, and I don't consider "viewport subwindow" to be a true window change.  If it is considered a window change then please update the documentation to reflect this behavior along with what UI classes trigger this.

**Comment by YeldhamDev** [comment]
Can't reproduce this in GNU/Linux.

**Comment by mouring** [comment]
Still can reproduce using 4.0a12 on Mac using https://www.dropbox.com/s/lqzlku0u4wbqfck/example.zip example.

**Comment by YeldhamDev** [comment]
OK, I managed to reproduce it. It only happens if `embed_subwindows` is enabled.

**Comment by akien-mga** [comment]
Needs a new MRP, the Dropbox links expired. You should be able to attach a ZIP below 10 MB here, which should be big enough to demonstrate a GUI issue (make sure to remove the `.godot` folder so that it doesn't include the heavy shader cache).

**Comment by mouring** [comment]
Regretfully I wasn't able to attach it in July, and now that stuff has been move around I no longer have this zip file, and the project I extracted it from is complete broken as I've not been keeping up with beta/RC.  And since I've paused my coding for a few months to reorder things I regretfully will not get back to this.

I resolved my issue using sriramun's suggestion above.  Which is technically the better way.

So feel free to close this if you wish.
